### PR TITLE
*: support log

### PR DESCRIPTION
**Pull request by BusyJay**
When we want to debug grpc, its log will output to stderr which is not useful for deployment. So this pr provides a function that can redirect grpc log to the standard rust log.

### CODE REVIEW

**Comment by siddontang** [comment]
Must user initialize a log to use gRPC lib now?

**Comment by BusyJay** [comment]
No. User can use `redirect_log` to integrate grpc log into their log system. Or do nothing, the library will work just the same way as before.

**Comment by siddontang** [comment]
LGTM

**Comment by BusyJay** [comment]
ping

File: src/log_util.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    // can't panic.
+    let file_str = unsafe { CStr::from_ptr(args.file).to_str().unwrap() };
+    let line = args.line as u32;
+
+    // use hidden API for now, will
+    // TODO: use public API once available.
+    let location = LogLocation {
+        __module_path: module_path!(),
```
Comment (overvenus) [inline src/log_util.rs]: Is `__module_path` always `grpcio::log_util`?

Comment (BusyJay) [reply src/log_util.rs]: For now yes.

File: src/log_util.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        return;
+    }
+
+    // can't panic.
+    let file_str = unsafe { CStr::from_ptr(args.file).to_str().unwrap() };
+    let line = args.line as u32;
+
+    // use hidden API for now, will
+    // TODO: use public API once available.
+    let location = LogLocation {
```
Comment (overvenus) [inline src/log_util.rs]: `Loglocation` has been removed (https://github.com/rust-lang-nursery/log/pull/162).

The latest `log` v0.3.8 is release in May 24, and differs  greatly from the master. Any new release of log will break grpcio's build. grpcio and log are tightly coupled if this PR is merged. Crates that use both log and grpcio may fail to build, if log and grpcio are mismatch.

File: Cargo.toml [hunk 3]
```diff
@@ -16,6 +16,7 @@ grpcio-sys = { path = "grpc-sys", version = "0.1.0" }
 libc = "0.2"
 futures = "^0.1.15"
 protobuf = { version = "1.2", optional = true }
+log = "0.3"
```
Comment (overvenus) [inline Cargo.toml]: Hopefully, next release of log starts from “0.4”.

File: src/log_util.rs (hunk 2 above)
Comment (BusyJay) [reply src/log_util.rs]: Yes. Already consider this. See TODO at L41 and rust-lang-nursery/log#223.

**Pull request merged**